### PR TITLE
Transfer certificate api endpoint

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

**Comment by tnickelsen** [comment]
fixes #36

File: src/Protos/wallet.proto [hunk 1]
```diff
@@ -30,28 +31,37 @@ service WalletService {
     // Events should also be able to be emitted.. RPC or event bus?
 }
 
+message CreateReceiverDepositEndpointRequest {
+    WalletDepositEndpoint WalletDepositEndpoint = 1;
+    string reference = 2;
+}
+
+message CreateReceiverDepositEndpointResponse {
+    project_origin.common.v1.Uuid ReceiverId = 1;
```
Comment (duizer) [inline src/Protos/wallet.proto:40]: Has there been considerations about creating a message just for the ReceiverId? A bit similar to the FederatedStreamId. This will make the ReceiverId strongly typed and clients will (hopefully) make fewer mistakes.

File: src/ProjectOrigin.WalletSystem.Server/ProjectOrigin.WalletSystem.Server.csproj [hunk 2]
```diff
@@ -14,6 +14,7 @@
     <PackageReference Include="NBitcoin" Version="7.0.25" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="7.0.4" />
     <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="7.0.0" />
+    <PackageReference Include="Portable.BouncyCastle" Version="1.9.0" />
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.Server/ProjectOrigin.WalletSystem.Server.csproj:17]: Why is this added?

File: src/ProjectOrigin.WalletSystem.IntegrationTests/ReceiveSliceTests.cs [hunk 3]
```diff
@@ -36,8 +36,8 @@ public async Task ReceiveSlice()
                     Registry = registryName,
                     StreamId = new Common.V1.Uuid() { Value = certId.ToString() },
                 },
-                WalletSectionPublicKey = ByteString.CopyFrom(section.PublicKey.Export()),
-                WalletSectionPosition = 2,
+                WalletDepositEndpointPublicKey = ByteString.CopyFrom(section.PublicKey.Export()),
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/ReceiveSliceTests.cs:39]: The entity stored in the database is still called "WalletSection", hence the variable `section`. Do we want to apply the rename all the way down or do we stop at the API level?

Comment (duizer) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/ReceiveSliceTests.cs:39]: Will be done in https://github.com/project-origin/wallet/issues/45

File: src/Protos/wallet.proto [hunk 4]
```diff
@@ -30,28 +31,37 @@ service WalletService {
     // Events should also be able to be emitted.. RPC or event bus?
 }
 
+message CreateReceiverDepositEndpointRequest {
+    WalletDepositEndpoint WalletDepositEndpoint = 1;
+    string reference = 2;
```
Comment (duizer) [inline src/Protos/wallet.proto:36]:
```suggestion
    string Reference = 2;
```

File: src/Protos/wallet.proto (hunk 1 above)
Comment (duizer) [reply src/Protos/wallet.proto:40]: @MartinSchmidt what do you think?

**Review by duizer — APPROVED** [review]
